src/app/components/chat/window/output/AppChatWindowOutputItem.vue

```diff
@@ -0,0 +1,530 @@
+<script lang="ts">
+import { transparentize } from 'polished';
+import { computed, inject, PropType, reactive, ref, toRefs } from 'vue';
+import { RouterLink } from 'vue-router';
+import { ContentRules } from '../../../../../_common/content/content-editor/content-rules';
+import { ContentOwnerParentBounds } from '../../../../../_common/content/content-owner';
+import AppContentViewer from '../../../../../_common/content/content-viewer/content-viewer.vue';
+import { formatDate } from '../../../../../_common/filters/date';
+import AppJolticon from '../../../../../_common/jolticon/AppJolticon.vue';
+import { ModalConfirm } from '../../../../../_common/modal/confirm/confirm-service';
+import { vAppObserveDimensions } from '../../../../../_common/observe-dimensions/observe-dimensions.directive';
+import { Popper } from '../../../../../_common/popper/popper.service';
+import AppPopper from '../../../../../_common/popper/popper.vue';
+import { DefaultTheme } from '../../../../../_common/theme/theme.model';
+import { useThemeStore } from '../../../../../_common/theme/theme.store';
+import { vAppTooltip } from '../../../../../_common/tooltip/tooltip-directive';
+import AppTranslate from '../../../../../_common/translate/AppTranslate.vue';
+import { $gettext } from '../../../../../_common/translate/translate.service';
+import { ChatStore, ChatStoreKey } from '../../chat-store';
+import {
+	removeMessage as chatRemoveMessage,
+	setMessageEditing,
+	userCanModerateOtherUser,
+} from '../../client';
+import { ChatMessage } from '../../message';
+import { ChatRoom } from '../../room';
+import AppChatUserPopover from '../../user-popover/user-popover.vue';
+import AppChatWindowOutputItemDetails from './AppChatWindowOutputItemDetails.vue';
+
+export interface ChatMessageEditEvent {
+	message: ChatMessage;
+}
+</script>
+
+<script lang="ts" setup>
+const props = defineProps({
+	message: {
+		type: Object as PropType<ChatMessage>,
+		required: true,
+	},
+	room: {
+		type: Object as PropType<ChatRoom>,
+		required: true,
+	},
+});
+
+const { message, room } = toRefs(props);
+
+const chatStore = inject<ChatStore>(ChatStoreKey)!;
+const { theme, isDark } = useThemeStore();
+
+const displayRules = new ContentRules({ maxMediaWidth: 400, maxMediaHeight: 300 });
+
+const root = ref<HTMLElement>();
+const itemWrapper = ref<HTMLElement>();
+
+const contentViewerBounds: ContentOwnerParentBounds = reactive({
+	width: ref(200),
+});
+
+const chat = computed(() => chatStore.chat!);
+
+// Use the form/page/user theme, or the default theme if none exist.
+const actualTheme = computed(() => theme.value ?? DefaultTheme);
+
+const isEditing = computed(() => chat.value.messageEditing === message.value);
+
+/** The background-color for chat items that are being edited. */
+const isEditingColor = computed(() => {
```

Comment:
what do you think of finally making a new "theme-primary" var that would be highlight in dark and backlight in light? We can then do all of this in CSS with the rgb vars.